### PR TITLE
[DailyWire] Add extractor

### LINKED ISSUE #3139: DailyWire Support Request
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

USA

### Example URLs

https://www.dailywire.com/episode/1-fauci
https://www.dailywire.com/videos/the-hyperions
https://www.dailywire.com/episode/ep-124-bill-maher

### Description

It seems that the site does not utilize any DRM, as I am able to use my personal DW account, grab the m3u8 from my browser's network tools, and stream/download directly with the m3u8 file (with the appropriate auth/cookie (not sure exactly what's included in the m3u8 file address).

### Verbose log

```shell
kevin@linuxserver:~/youtube-dl/temp-downloads$ yt-dlp --version
2022.03.08.2
kevin@linuxserver:~/youtube-dl/temp-downloads$ yt-dlp -vU https://www.dailywire.com/episode/ep-124-bill-maher
[debug] Command-line config: ['-vU', 'https://www.dailywire.com/episode/ep-124-bill-maher']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.2 [e491d06d3]
[debug] Python version 3.6.9 (CPython 64bit) - Linux-4.15.0-171-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: ffmpeg 4.3.2-0york0, ffprobe 4.3.2-0york0
[debug] Optional libraries: brotli, Cryptodome, secretstorage, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.2
yt-dlp is up to date (2022.03.08.2)
[debug] [generic] Extracting URL: https://www.dailywire.com/episode/ep-124-bill-maher
[generic] ep-124-bill-maher: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] ep-124-bill-maher: Downloading webpage
[generic] ep-124-bill-maher: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.dailywire.com/episode/ep-124-bill-maher
Traceback (most recent call last):
  File "/home/kevin/.local/lib/python3.6/site-packages/yt_dlp/YoutubeDL.py", line 1389, in wrapper
    return func(self, *args, **kwargs)
  File "/home/kevin/.local/lib/python3.6/site-packages/yt_dlp/YoutubeDL.py", line 1459, in __extract_info
    ie_result = ie.extract(url)
  File "/home/kevin/.local/lib/python3.6/site-packages/yt_dlp/extractor/common.py", line 617, in extract
    ie_result = self._real_extract(url)
  File "/home/kevin/.local/lib/python3.6/site-packages/yt_dlp/extractor/generic.py", line 3998, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.dailywire.com/episode/ep-124-bill-maher
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

Add support for www.dailywire.com, only support 'episode', 'videos', and 'podcast'. Right now, this PR didn't include login function.

EDIT-2: The certificate error only appear at `yt-dlp` in my computer. Browser(firefox) and `curl` return html without any error. However, this is work fine in my mobile phone

Close #3139

### CODE REVIEW

**Comment by pukkandan** [comment]
> PS : This PR test are expected to failed as the url always give certificate error: urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)>, needed to pass --no-check-certificate

Is that with or without `certifi`?

**Comment by HobbyistDev** [comment]
It's odd, the url is working. Sorry if i bother you, maybe because my ISP.

Here's the log (working):
```
(python_proj) ~/.../python_proj/yt-dlp $ python -m yt_dlp -s -v "https://www.dailywire.com/episode/ep-124-bill-maher"
[debug] Command-line config: ['-s', '-v', 'https://www.dailywire.com/episode/ep-124-bill-maher']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 8e31fbcdf
[debug] Python version 3.10.4 (CPython 32bit) - Linux-3.18.14-14381225-QB20734933-armv8l-with-libc
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2022.05.18.1, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] [DailyWire] Extracting URL: https://www.dailywire.com/episode/ep-124-bill-maher
[DailyWire] ep-124-bill-maher: Downloading webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] cl0ngbaalplc80894sfdo9edf: Downloading 1 format(s): 0
```

**Comment by HobbyistDev** [comment]
The error came again but only in my computer and not in my mobile phone. It's very odd only occur at `yt-dlp`
`curl` and browser give normal html not have any ssl error

Here's the log:
```
(python_proj2) D:\user\Computer\python_proj\python_proj2\yt-dlp>python -m yt_dlp -v -s "https://www.dailywire.com/episode/1-fauci"
[debug] Command-line config: ['-v', '-s', 'https://www.dailywire.com/episode/1-fauci']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 8e31fbcdf
[debug] Python version 3.10.4 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1-full_build-www.gyan.dev (setts), ffprobe 5.0.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2022.05.18.1, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] [DailyWire] Extracting URL: https://www.dailywire.com/episode/1-fauci
[DailyWire] 1-fauci: Downloading webpage
ERROR: [DailyWire] Unable to download webpage: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)> (caused by URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)')))
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 642, in extract
    ie_result = self._real_extract(url)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\dailywire.py", line 57, in _real_extract
    slug, episode_info = self._get_json(url, 'episode_name')
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\dailywire.py", line 18, in _get_json
    webpage = self._download_webpage(url, slug)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 1045, in _download_webpage
    return self.__download_webpage(url_or_request, video_id, note, errnote, None, fatal, *args, **kwargs)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 996, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 828, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 785, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\urllib\request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 1037, in _send_output
    self.send(msg)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 975, in send
    self.connect()
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\http\client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\ssl.py", line 512, in wrap_socket
    return self.sslsocket_class._create(
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\ssl.py", line 1070, in _create
    self.do_handshake()
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\ssl.py", line 1341, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\extractor\common.py", line 767, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\YoutubeDL.py", line 3656, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\urllib\request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "D:\user\Computer\python_proj\python_proj2\yt-dlp\yt_dlp\utils.py", line 1443, in https_open
    return self.do_open(
  File "D:\user\Computer\winpython\WPy64-31040\python-3.10.4.amd64\lib\urllib\request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)>
```
Here's the curl log:
```
(python_proj2) D:\user\Computer\python_proj\python_proj2\yt-dlp>curl -v "https://www.dailywire.com/episode/1-fauci"
*   Trying 76.76.21.22:443...
* Connected to www.dailywire.com (76.76.21.22) port 443 (#0)
* schannel: disabled automatic use of client certificate
* schannel: ALPN, offering http/1.1
* schannel: ALPN, server accepted to use http/1.1
> GET /episode/1-fauci HTTP/1.1
> Host: www.dailywire.com
> User-Agent: curl/7.79.1
> Accept: */*
>
* schannel: failed to decrypt data, need more data
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Type: text/html; charset=utf-8
< Connection: keep-alive
< access-control-allow-origin: https://public.dailywire.com
< x-matched-path: /episode/[slug]
< cache-control: public, max-age=0, must-revalidate
< content-length: 85450
< date: Fri, 17 Jun 2022 01:38:37 GMT
< x-powered-by: Next.js
< etag: "14dca-EhSrCGO74j+Ok3i9kdeO5xnBlTg"
< x-vercel-cache: HIT
< age: 477
< server: Vercel
< x-vercel-id: sin1::iad1::h4jld-1655430395630-a1bb95705a88
< strict-transport-security: max-age=63072000
... return html of web correctly
```
In my browser (firefox) give information of certificate like this:

```
Validity:
Not Before : Fri, 13 May 2022 19:30:42 GMT
Not After : Thu, 11 Aug 2022 19:30:41 GMT
```

Should i open new issue for this ?

**Comment by pukkandan** [comment]
> The error came again but only in my computer and not in my mobile phone. It's very odd only occur at yt-dlp.
curl and browser give normal html and not have any ssl error or expired certificate

I can't reproduce, and honestly have no clue what the issue could be

File: yt_dlp/extractor/dailywire.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    def _real_extract(self, url):
+        slug, episode_info = self._get_json(url, 'episode_name')
+        formats, subtitle = [], {}
+        for segment in episode_info.get('segments') or episode_info['videoUrl']:
+            subs = {}
+            if segment.get('audio') in ('Access Denied', None) and segment.get('video') in ('Access Denied', None):
+                continue
+            if segment.get('video') not in ('Access Denied', None):
+                format_, subs = self._extract_m3u8_formats_and_subtitles(
+                    segment.get('audio') or segment.get('video'), slug
```
Comment (pukkandan) [inline yt_dlp/extractor/dailywire.py:65]: What is the intend here? If video is not "Access denied", use audio?

**Comment by pukkandan** [comment]
1. Fix tests - Did you even run them in the first place??
2. You can extract timestamp, modified_timestamp, release_timestamp, series, rating

File: yt_dlp/extractor/dailywire.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+class DailyWireIE(DailyWireBaseIE):
+    _VALID_URL = r'https?://(?:www\.)dailywire(?:\.com)/(?P<sites_type>episode|videos)/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://www.dailywire.com/episode/1-fauci',
+        'info_dict': {
+            'id': 'ckzsl50xnqpy30850in3v4bu7',
+            'ext': 'mp4',
+            'display_id': '1-fauci',
+            'title': '1. Fauci',
+            'description': 'Lee explores the history of the politics surrounding Covid-19, and reveals shocking details about both Anthony Fauci\'s role in hiding critical information from the American people as well as worrisome unanswered questions surrounding the origins of the virus.',
```
Comment (pukkandan) [inline yt_dlp/extractor/dailywire.py:33]: Use md5

tip: If you run a wrong test-case, it will tell you how to fix itself

**Comment by HobbyistDev** [comment]
I am sorry for dirty commit because i can't run the test in my machine so i will push small commit here and then test the `_TESTS` in other machine

**Comment by pukkandan** [comment]
That's a weird workflow, but okay. Just ping me when you are done

File: yt_dlp/extractor/dailywire.py (hunk 2 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/dailywire.py]: Fixed

**Pull request merged**